builder/builder.sh

```diff
@@ -0,0 +1,26 @@
+#!/bin/bash
+
+set -e
```

Comment:
Suggestion:

```suggestion
set -eu
```

Always best to start like this in a brand new script. Adding `-u` is much harder to add once the script becomes non-trivial.

Reply:
thanks for flagging that. i had to look up what `set -u` means:

> Treat unset variables and parameters other than the special parameters "@" and "*" as an error when performing parameter expansion.

completely agree trying to retrofit into a large gnarly important bash script could be unpleasant

Reply:
`set -eu` is my default. Some people like `-o pipefail`, but I personally don't. [Google's shell style guide](https://google.github.io/styleguide/shellguide.html) and shellcheck are good resources, if you are interested.